components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14757,6 +14757,37 @@ public String addEndpointRegistry(EndpointRegistryInfo endpointRegistry, int ten
         return uuid;
     }
 
+    /**
+     * Update an existing endpoint registry.
+     *
+     * @param registryId uuid of the endpoint registry
+     * @param endpointRegistry EndpointRegistryInfo object with updated details
+     * @return uuid of the updated endpoint registry
+     * @throws APIManagementException if unable to update the endpoint registry
+     */
+    public String updateEndpointRegistry(String registryId, EndpointRegistryInfo endpointRegistry) throws
+            APIManagementException {
+
+        String query = SQLConstants.UPDATE_ENDPOINT_REGISTRY_SQL;
+        String uuid = UUID.randomUUID().toString();
```

Comment:
We don't need to generate uuid when updating the Registry. Shall we remove it?

Reply:
ack